### DISCUSSION TITLE
What are the supported auth types for a plugin?

### DISCUSSION THREAD

**Discussion by eeadie** [microsoft/semantic-kernel#5884] [Q&A] [answered]
Hi,

What are the supported auth types for a plugin?

![Screenshot 2024-04-16 140052](https://github.com/microsoft/semantic-kernel/assets/23075647/811020ac-9052-4d6a-8a91-ca5d0eeeb413)

I have tried setting auth type to "bearer" however this results in the plugin failing to load using Chat Copilot:

![Screenshot 2024-04-16 140313](https://github.com/microsoft/semantic-kernel/assets/23075647/2a8cf67e-5eba-44da-984c-de1d521f8421)

**Comment by sophialagerkranspandey** [comment]
Within the Semantic Kernel SDK any auth type can be supported. Within Chat Copilot only a few types are supported, such as Key based. @teresaqhoang can you please help with what types are supported?

**Comment by teresaqhoang** [comment]
Hi @eeadie , Chat Copilot only supports plugins requiring:
- No Auth https://platform.openai.com/docs/plugins/authentication/no-authentication
- or User-Level auth https://platform.openai.com/docs/plugins/authentication/user-level

"None" should be valid, were you getting errors before you changed it?

**Comment by eeadie** [comment]
@teresaqhoang the plugin loads as expected when the auth is set to none. I was investigating what other types of auth was possible for the plugin. I noticed in Chat Copilot the Jira plugin allows the user to enter a personal access token. Do you know what auth/plugin settings are required to allow the entering of a personal access token when enabling the plugin in Chat Copilot?

![Screenshot 2024-04-17 085437](https://github.com/microsoft/semantic-kernel/assets/23075647/afb82ea5-5fab-4e0a-9875-20a644493cb3)

**Comment by teresaqhoang** [comment, marked as answer]
Hi @eeadie, 

Apologies, Chat Copilot is currently in maintenance-only mode, so we haven't made any updates since OpenAI deprecated manifest files for plugins, but our plugins still require the manifest file. 

If you set the "auth" field in your manifest file (ai-plugin.json) with "type" as `user_http` and "authorization_type" as `Bearer`, you should be able to use a PAT.

```
...
"auth": {
  "type": "user_http",
  "authorization_type": "bearer",
},
```

Please keep in mind that chat copilot is meant as a playground for plugins, so use this auth option if you are developing an “unverified app” that you will have to distribute manually.

**Comment by eeadie** [comment]
@teresaqhoang many thanks for your response, the configuration worked :)
